benchmark_v2/str_locality_benchmark.py

```diff
@@ -0,0 +1,43 @@
+import arkouda as ak
+import pytest
+
+OPS = {
+    "Hashing": lambda x: x.hash(),
+    "Regex_Search": lambda x: x.contains(r"\d{3,5}\.\d{5,8}", regex=True),
+    "Casting": lambda x: ak.cast(x, ak.float64),
+    "Scalar_Compare": lambda x: (x == "5.5")
+}
+
+# Good - generates random Strings object with "good" locality
+# poor - generates a sorted Strings object with "poor" locality
+LOCALITY = {"Good", "Poor"}
+
+
+def _generate_data(loc):
+    N = pytest.prob_size * ak.get_config()["numLocales"]
+    prefix = ak.random_strings_uniform(minlen=1, maxlen=16, size=N, seed=pytest.seed, characters="numeric")
```

Comment:
It would be nice to use the same random data between parameters instead of generating new each time. I'm not sure if it's possible in this case

Reply:
@pierce314159 - I created a global for random data and another for sorted data to handle both cases. I ran the benchmark and save the json with the name of the Strings object used. I verified that the name was identical for all poor locality tests and for all good locality tests. 

The only change I may want to make is to set both objects at once and only return the one needed. This would ensure that the data is the same just sorted or not. I think this would be good to do so I will look at that as well.

Reply:
Decided to make the further test I mentioned previously. The code will now return the value desired if the objects were already created. Otherwise, it creates both objects so that the same data is random and sorted. This should ensure that the current benchmark is as close to the previous as possible.